### ISSUE TITLE
ERROR-level log spam when non-leader node restarts during writes (view-building)

### ISSUE THREAD

**Issue by nyh** [scylladb/scylladb#17290]
The dtest `materialized_views_test.py::TestMaterializedViews::test_mv_populating_from_existing_data_during_node_restart` when run with tablets (`--tablets`) passes in the actual test, but then fails because of unrecognized ERROR-level messages in the log. 

We need to decide if these ERROR messages are fine and should be ignored by the test, or whether they should be downgraded to WARN or even lower.

That test starts a cluster of four nodes, fills a base table, and then adds a view (which causes view building - and in particular writes - to happen in the background) while in parallel **restarts** the second node. After the second node restarts, it waits to check the view finishes building and contains the right data. As I mentioned, the building and data is correct and the test would pass, except for the errors:

First, the node being restarted reports the following error during shutdown. I don't think this should be an ERROR-level message?
```
INFO  2024-02-12 15:22:10,829 [shard 0:main] init - Shutting down group 0 usage in local storage
INFO  2024-02-12 15:22:10,829 [shard 0:strm] raft_topology - raft_state_monitor_fiber aborted with seastar::abort_requested_exception (abort requested)
INFO  2024-02-12 15:22:10,829 [shard 0:strm] raft_topology - cleanup fiber aborted
ERROR 2024-02-12 15:22:10,829 [shard 0: gms] raft_topology - raft_topology_cmd failed with: raft::request_aborted (Request is aborted by a caller)
ERROR 2024-02-12 15:22:10,829 [shard 0: gms] raft_topology - raft_topology_cmd failed with: raft::request_aborted (Request is aborted by a caller)
INFO  2024-02-12 15:22:10,829 [shard 0:main] init - Shutting down group 0 usage in local storage was successful
```

Second, the first node, who the logs show is the group 0 leader ("raft_group0 - gaining leadership"), reports the following ERRORs in the log about the second node going down:
```
INFO  2024-02-12 15:22:08,767 [shard 0: gms] gossip - Got shutdown message from 127.0.76.2, received_generation=1707744102, local_generation=1707744102
INFO  2024-02-12 15:22:08,769 [shard 0: gms] gossip - InetAddress 127.0.76.2 is now DOWN, status = shutdown
ERROR 2024-02-12 15:22:08,806 [shard 0:strm] raft_topology - drain rpc failed, proceed to fence old writes: std::runtime_error (raft topology: exec_global_command(barrier_and_drain) failed with seastar::rpc::closed_error (connection is closed))
```
and later a bunch of errors like
```
ERROR 2024-02-12 15:22:10,768 [shard 0:strm] raft_topology - barrier for tablet b89e4460-c9a9-11ee-b7e1-0155ece4f661:0 failed: seastar::broken_promise (broken promise)
```
and finally
```
ERROR 2024-02-12 15:22:10,768 [shard 0:strm] raft_topology - topology change coordinator fiber got error std::runtime_error (raft topology: exec_global_command(barrier) failed with seastar::rpc::closed_error (connection is closed))
INFO  2024-02-12 15:22:11,630 [shard 0:main] raft_group_registry - marking Raft server 43d4c11a-bdce-4f7b-b304-da61beab4fbf as dead for raft groups
```

Again, it seems that all these messages just report the rather **normal** occurance of a node going down, and isn't a real error that a user needs to be concerned with, so perhaps they should be downgraded to WARN.

If we don't downgrade these messages, we need to ignore these messages in the test (or all tests??)

**Comment by nyh** [comment]
@tgrabiec I could use your advice on what to do here. Is this a known problem? Would you consider these ERROR messages to be acceptable - and the test needs to learn to ignore them, or these messages sound too scary and need to be downgraded?

**Comment by tgrabiec** [comment]
With background load balancing it's expected that when you down a node you will get ERRORs like this. 

We can either:
  1) quiesce load balancer before downing a node (@michoecho mentioned he wanted to add API to do this for his work). 
  2) add all topology-related errors as expected everywhere
  3) demote such errors to WARN

Doing (1) in all tests would reduce coverage, in reality balancing may run concurrently with other things so it's good to exercise this. So seems the least attractive option unless necessary.

The problem with (3) is that the errors are logged in a generic place (e.g. "topology change coordinator fiber got error ...") which doesn't know if the failure is benign (node down) or something more serious and permanent.

**Comment by tchaikov** [comment]
just for SEO: the same symptom is observed when testing `replace_address_test.py::TestReplaceAddress::test_replace_node_same_ip` with `--tablets`.

**Comment by nyh** [comment]
I want to do @tgrabiec's (4) - accept these errors in tests that bring a node down - but the problem is that:
1. The error is too vague to ignore it in all tests (this is why @tgrabiec asked not to choose option 1).
2. I can ignore it in specific tests, including test_mv_populating_from_existing_data_during_node_restart which I found and test_replace_node_same_ip that @tchaikov found above - but maybe there are a dozen other tests we need to do this in. I don't even know how to find them :-(

**Comment by nyh** [comment]
By the way, @tgrabiec I honestly think this error message is wrong, and should be fixed by I am not familiar enough with this code to suggest how to fix it, let alone fix it, so I'll fix the test issue by ignoring this error and will suggest that someone else (@kbr-scylla ?) later consider changing or dropping this error message:

Basically, the user will see in the log messages like:

```
ERROR 2024-03-19 17:32:05,038 [shard 0: gms] raft_topology - raft_topology_cmd failed with: raft::request_aborted (Request is aborted by a caller)
```
and
```
ERROR 2024-03-19 17:34:49,969 [shard 0:strm] raft_topology - drain rpc failed, proceed to fence old writes: std::runtime_error (raft topology: exec_global_command(barrier_and_drain) failed with seastar::rpc::closed_error (connection is closed))
```
Whenever shutting down a node, or perhaps just when some tablet happens to be migration at the same time, or I'm not sure exactly what. But these error message are completely obscure, and if I don't understand what either of them means, there is no chance that end-users will understand it when they see these error messages, and know if there's a problem or not.

If before Raft nodes did not log ERROR messages when they noticed that they or other nodes are going down, we shouldn't start this practice in Raft.

@kbr-scylla if you agree with me on this, please open a new issue for it because I plan to close this one once I patch the test to ignore the error message, and I don't know exactly what to write in a new issue.

**Comment by nyh** [comment]
I just got another failure in the same test:
```
ERROR 2024-03-19 17:41:45,281 [shard 0:strm] stream_session - [Stream #309ba3d0-e607-11ee-96a4-f0ecd37a4280] Failed to handle STREAM_MUTATION_FRAGMENTS (receive and distribute phase) for ks=ks, cf=tm_table, peer=127.0.19.4: seastar::nested_exception: seastar::rpc::stream_closed (rpc stream was closed by peer) (while cleaning up after seastar::rpc::stream_closed (rpc stream was closed by peer))
```
This time it seems that the table's streaming was interrupted in the middle when the node went down. Which, again, is legitimate and isn't an error. I'll have to ignore that one too :-(

**Comment by nyh** [comment]
The plot thickens :-(

After I add all these error message exclusions from the test, the test passes, finally. But... then, in addition to the passing test, pytest **also** gives an error when shutting down the fixtures,
```
dtest_setup.check_errors_all_nodes(exclude_errors=exclude_errors)
```

because apparently in conftest.py we have yet another call of
```
dtest_setup.check_errors_all_nodes(exclude_errors=exclude_errors)
```
and this one doesn't have the exclusions.

I'll try to figure out how to set this "exclude_errors" on this test (it seems I need a marker), but... it seems that this error can appear on *any* test where a node is shut down during background tablet migration.

So @fruch @bhalevy  I don't know what the proper fix is. I want to fix this specific MV dtest which my assignment says I should fix, but I don't want other people to see the same error again in 17 other tests when enabling tablets :-(

**Comment by fruch** [comment]
> The plot thickens :-(
> 
> After I add all these error message exclusions from the test, the test passes, finally. But... then, in addition to the passing test, pytest **also** gives an error when shutting down the fixtures,
> ```
> ERROR 2024-03-19 17:53:43,777 [shard 0: gms] raft_topology - raft_topology_cmd failed with: raft::request_aborted (Request is aborted by a caller)
> ```
> 
> because apparently in conftest.py we have yet another call of
> ```
> dtest_setup.check_errors_all_nodes(exclude_errors=exclude_errors)
> ```
> and this one doesn't have the exclusions.
> 
> I'll try to figure out how to set this "exclude_errors" on this test (it seems I need a marker), but... it seems that this error can appear on *any* test where a node is shut down during background tablet migration.

It can be done in multiple ways, marker is one of the most recent options for it introduced by @tchaikov 

See all those tests
https://github.com/search?q=repo%3Ascylladb%2Fscylla-dtest+self.ignore_log_patterns&type=code

On how to control the check of logs in teardown 

> 
> So @fruch @bhalevy  I don't know what the proper fix is. I want to fix this specific MV dtest which my assignment says I should fix, but I don't want other people to see the same error again in 17 other tests when enabling tablets :-(

**Comment by nyh** [comment]
> > The plot thickens :-(
> > After I add all these error message exclusions from the test, the test passes, finally. But... then, in addition to the passing test, pytest **also** gives an error when shutting down the fixtures,
> > ```
> > ERROR 2024-03-19 17:53:43,777 [shard 0: gms] raft_topology - raft_topology_cmd failed with: raft::request_aborted (Request is aborted by a caller)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > because apparently in conftest.py we have yet another call of
> > ```
> > dtest_setup.check_errors_all_nodes(exclude_errors=exclude_errors)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > and this one doesn't have the exclusions.
> > I'll try to figure out how to set this "exclude_errors" on this test (it seems I need a marker), but... it seems that this error can appear on _any_ test where a node is shut down during background tablet migration.
> 
> It can be done in multiple ways, marker is one of the most recent options for it introduced by @tchaikov

Oh, thanks I missed the self.ignore_log_patterns trick :-( I used the marker but it was buggy so I had to fix it, see the first patch in https://github.com/scylladb/scylla-dtest/pull/4082.

Anyway, that PR makes this single test that used to fail, pass. I don't know how many other tests fail like this across the test suite (I am only looking at MV-related tests right now), but I'm sure there are and I don't like this solution - but without agreement from the Tablets and Raft developers that these error messages are wrong, it's the only thing I can do :-(

**Comment by bhalevy** [comment]
The advantage of ignore_log_patterns is that message can be added to it right before the event that might trigger them is executed, so if they happen before they are expected they's still fail the test when it checks for unexpected errors on exit.
Not a big deal.
Also, they can be tailored based on more intricate conditions like gossip vs. raft, streaming vs. rbno, etc.

**Comment by mykaul** [comment]
I'm unsure why is @nyh assigned to this?

**Comment by nyh** [comment]
> I'm unsure why is @nyh assigned to this?

After I couldn't get consensus for stopping to print the spammy error messages, I created a PR to fix the test to ignore them - https://github.com/scylladb/scylla-dtest/pull/4082 - but it didn't go in either. I'll take a look again now why this happened, I forgot to chase it.

**Comment by nyh** [comment]
Pushed v2:
1. Rebased
2. Use only the single error-ignoring mechanism, check_errors_all_nodes(), which the existing test used. The reviewers didn't like my attempts to use two different mechanisms for the test itself and the post-test shutdown - the parameters for check_errors_all_nodes() are also ignored during the shutdown - but that only works if there are _not_ regular expressions.

**Comment by avikivity** [comment]
If you fix a test to ignore errors, you should also fix the users to ignore the same errors. We should try to have higher quality error messages.

**Comment by fruch** [comment]
> If you fix a test to ignore errors, you should also fix the users to ignore the same errors. We should try to have higher quality error messages.

now I have SCT run with 1000x of such a messge:
```
2024-08-03 05:00:54.807 <2024-08-03 05:00:54.714>: (DatabaseLogEvent Severity.ERROR) period_type=one-time event_id=d418319d-5091-4cbe-aa16-b2f1d40f80de: type=RUNTIME_ERROR regex=std::runtime_error line_number=33985 node=longevity-50gb-12h-master-db-node-a991fff7-1
2024-08-03T05:00:54.714+00:00 longevity-50gb-12h-master-db-node-a991fff7-1      !ERR | scylla[9237]:  [shard 0: gms] raft_topology - drain rpc failed, proceed to fence old writes: std::runtime_error (raft topology: exec_global_command(barrier_and_drain) failed with seastar::rpc::closed_error (connection is closed))
```

**Comment by kbr-scylla** [comment]
@fruch was this also during some node restart? Or some different scenario? I see this message is being printed once every second:
```
Aug 03 04:46:59.746508 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] raft_topology - entered `tablet migration` transition state
Aug 03 04:46:59.746558 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] load_balancer - Examining DC eu-west (shuffle=false, balancing=true)
Aug 03 04:46:59.746596 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] load_balancer - Node 240b237d-48eb-4cbe-91fc-aeb9bef4f828: rack=1c avg_load=13.25, tablets=106, shards=8, state=normal
Aug 03 04:46:59.746601 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] load_balancer - Node 4907ae4f-a18c-4113-9a1f-30f36eafe580: rack=1c avg_load=9.142857142857142, tablets=128, shards=14, state=normal
Aug 03 04:46:59.746606 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] load_balancer - Node b9e84dc9-e8d4-4775-b1fb-46a1053ad63e: rack=1c avg_load=9.142857142857142, tablets=128, shards=14, state=normal
Aug 03 04:46:59.746839 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] load_balancer - target node: 4907ae4f-a18c-4113-9a1f-30f36eafe580, avg_load: 9.142857142857142, max: 13.25
Aug 03 04:46:59.748234 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] load_balancer - Not possible to achieve balance.
Aug 03 04:46:59.748319 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] load_balancer - Prepared 0 migrations in DC eu-west
Aug 03 04:46:59.748331 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] load_balancer - Table 09c184e0-5147-11ef-9682-3f23bab71129 with tablet_count=32 has an average tablet size of 6034263366
Aug 03 04:46:59.748471 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] load_balancer - Table f7fc49e0-514e-11ef-9682-3f23bab71129 with tablet_count=16 has an average tablet size of 0
Aug 03 04:46:59.748481 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] load_balancer - Table 87e89e50-514f-11ef-8b33-a0945668a88b with tablet_count=16 has an average tablet size of 0
Aug 03 04:46:59.748486 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] load_balancer - Table bb90ccf0-514f-11ef-9682-3f23bab71129 with tablet_count=16 has an average tablet size of 0
Aug 03 04:46:59.748490 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] load_balancer - Table 273dc300-514f-11ef-8b33-a0945668a88b with tablet_count=16 has an average tablet size of 0
Aug 03 04:46:59.748494 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] load_balancer - Table c88800f0-514e-11ef-9682-3f23bab71129 with tablet_count=16 has an average tablet size of 0
Aug 03 04:46:59.748498 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] load_balancer - Table 5756b060-514f-11ef-8b33-a0945668a88b with tablet_count=16 has an average tablet size of 0
Aug 03 04:46:59.748504 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] load_balancer - Prepared 0 migration plans, out of which there were 0 tablet migration(s) and 0 resize decision(s)
Aug 03 04:46:59.748514 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] raft_topology - executing global topology command barrier_and_drain, excluded nodes: {}
Aug 03 04:46:59.750419 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] raft_topology - drain rpc failed, proceed to fence old writes: std::runtime_error (raft topology: exec_global_command(barrier_and_drain) failed with seastar::rpc::closed_error (connection is closed))
Aug 03 04:46:59.751157 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] raft_topology - updating topology state: advance fence version to 71
Aug 03 04:46:59.787690 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] raft_topology - executing global topology command barrier, excluded nodes: {}
Aug 03 04:46:59.831638 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] raft_topology - barrier for tablet 09c184e0-5147-11ef-9682-3f23bab71129:29 failed: seastar::broken_promise (broken promise)
Aug 03 04:46:59.831675 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] raft_topology - barrier for tablet 09c184e0-5147-11ef-9682-3f23bab71129:19 failed: seastar::broken_promise (broken promise)
Aug 03 04:46:59.831690 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] raft_topology - barrier for tablet 273dc300-514f-11ef-8b33-a0945668a88b:14 failed: seastar::broken_promise (broken promise)
Aug 03 04:46:59.831703 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] raft_topology - barrier for tablet 5756b060-514f-11ef-8b33-a0945668a88b:15 failed: seastar::broken_promise (broken promise)
Aug 03 04:46:59.831717 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] raft_topology - barrier for tablet 5756b060-514f-11ef-8b33-a0945668a88b:12 failed: seastar::broken_promise (broken promise)
Aug 03 04:46:59.831730 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] raft_topology - barrier for tablet 5756b060-514f-11ef-8b33-a0945668a88b:14 failed: seastar::broken_promise (broken promise)
Aug 03 04:46:59.831743 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] raft_topology - barrier for tablet 5756b060-514f-11ef-8b33-a0945668a88b:1 failed: seastar::broken_promise (broken promise)
Aug 03 04:46:59.831755 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] raft_topology - barrier for tablet 5756b060-514f-11ef-8b33-a0945668a88b:0 failed: seastar::broken_promise (broken promise)
Aug 03 04:46:59.831780 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] raft_topology - topology change coordinator fiber got error std::runtime_error (raft topology: exec_global_command(barrier) failed with seastar::rpc::closed_error (connection is closed))
Aug 03 04:47:00.129267 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 1:comp] sstable - Rebuilding bloom filter /var/lib/scylla/data/keyspace1/standard1-09c184e0514711ef96823f23bab71129/me-3gid_0dab_2gy5s2uusi2lt4jxf2-big-Filter.db: resizing bitset from 173048 bytes to 197680 bytes. sstable origin: compaction
Aug 03 04:47:00.832274 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] raft_topology - entered `tablet migration` transition state
Aug 03 04:47:00.832328 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] load_balancer - Examining DC eu-west (shuffle=false, balancing=true)
Aug 03 04:47:00.832368 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] load_balancer - Node 240b237d-48eb-4cbe-91fc-aeb9bef4f828: rack=1c avg_load=13.25, tablets=106, shards=8, state=normal
Aug 03 04:47:00.832373 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] load_balancer - Node 4907ae4f-a18c-4113-9a1f-30f36eafe580: rack=1c avg_load=9.142857142857142, tablets=128, shards=14, state=normal
Aug 03 04:47:00.832377 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] load_balancer - Node b9e84dc9-e8d4-4775-b1fb-46a1053ad63e: rack=1c avg_load=9.142857142857142, tablets=128, shards=14, state=normal
Aug 03 04:47:00.832563 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] load_balancer - target node: 4907ae4f-a18c-4113-9a1f-30f36eafe580, avg_load: 9.142857142857142, max: 13.25
Aug 03 04:47:00.833876 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] load_balancer - Not possible to achieve balance.
Aug 03 04:47:00.833988 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] load_balancer - Prepared 0 migrations in DC eu-west
Aug 03 04:47:00.834000 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] load_balancer - Table 09c184e0-5147-11ef-9682-3f23bab71129 with tablet_count=32 has an average tablet size of 6034263366
Aug 03 04:47:00.834006 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] load_balancer - Table f7fc49e0-514e-11ef-9682-3f23bab71129 with tablet_count=16 has an average tablet size of 0
Aug 03 04:47:00.834010 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] load_balancer - Table 87e89e50-514f-11ef-8b33-a0945668a88b with tablet_count=16 has an average tablet size of 0
Aug 03 04:47:00.834014 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] load_balancer - Table bb90ccf0-514f-11ef-9682-3f23bab71129 with tablet_count=16 has an average tablet size of 0
Aug 03 04:47:00.834018 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] load_balancer - Table 273dc300-514f-11ef-8b33-a0945668a88b with tablet_count=16 has an average tablet size of 0
Aug 03 04:47:00.834022 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] load_balancer - Table c88800f0-514e-11ef-9682-3f23bab71129 with tablet_count=16 has an average tablet size of 0
Aug 03 04:47:00.834026 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] load_balancer - Table 5756b060-514f-11ef-8b33-a0945668a88b with tablet_count=16 has an average tablet size of 0
Aug 03 04:47:00.834032 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] load_balancer - Prepared 0 migration plans, out of which there were 0 tablet migration(s) and 0 resize decision(s)
Aug 03 04:47:00.834043 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] raft_topology - executing global topology command barrier_and_drain, excluded nodes: {}
Aug 03 04:47:00.836037 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] raft_topology - drain rpc failed, proceed to fence old writes: std::runtime_error (raft topology: exec_global_command(barrier_and_drain) failed with seastar::rpc::closed_error (connection is closed))
Aug 03 04:47:00.836455 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] raft_topology - updating topology state: advance fence version to 71
Aug 03 04:47:00.857915 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 1:comp] sstable - Rebuilding bloom filter /var/lib/scylla/data/keyspace1/standard1-09c184e0514711ef96823f23bab71129/me-3gid_0dac_0x09s2uusi2lt4jxf2-big-Filter.db: resizing bitset from 173048 bytes to 197680 bytes. sstable origin: compaction
Aug 03 04:47:00.865404 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] raft_topology - executing global topology command barrier, excluded nodes: {}
Aug 03 04:47:00.900405 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] raft_topology - barrier for tablet 09c184e0-5147-11ef-9682-3f23bab71129:29 failed: seastar::broken_promise (broken promise)
Aug 03 04:47:00.900442 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] raft_topology - barrier for tablet 09c184e0-5147-11ef-9682-3f23bab71129:19 failed: seastar::broken_promise (broken promise)
Aug 03 04:47:00.900455 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] raft_topology - barrier for tablet 273dc300-514f-11ef-8b33-a0945668a88b:14 failed: seastar::broken_promise (broken promise)
Aug 03 04:47:00.900467 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] raft_topology - barrier for tablet 5756b060-514f-11ef-8b33-a0945668a88b:15 failed: seastar::broken_promise (broken promise)
Aug 03 04:47:00.900480 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] raft_topology - barrier for tablet 5756b060-514f-11ef-8b33-a0945668a88b:12 failed: seastar::broken_promise (broken promise)
Aug 03 04:47:00.900493 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] raft_topology - barrier for tablet 5756b060-514f-11ef-8b33-a0945668a88b:14 failed: seastar::broken_promise (broken promise)
Aug 03 04:47:00.900505 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] raft_topology - barrier for tablet 5756b060-514f-11ef-8b33-a0945668a88b:1 failed: seastar::broken_promise (broken promise)
Aug 03 04:47:00.900517 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] raft_topology - barrier for tablet 5756b060-514f-11ef-8b33-a0945668a88b:0 failed: seastar::broken_promise (broken promise)
Aug 03 04:47:00.900539 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] raft_topology - topology change coordinator fiber got error std::runtime_error (raft topology: exec_global_command(barrier) failed with seastar::rpc::closed_error (connection is closed))
Aug 03 04:47:01.592549 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 1:comp] sstable - Rebuilding bloom filter /var/lib/scylla/data/keyspace1/standard1-09c184e0514711ef96823f23bab71129/me-3gid_0dac_5a44w2uusi2lt4jxf2-big-Filter.db: resizing bitset from 173048 bytes to 197680 bytes. sstable origin: compaction
Aug 03 04:47:01.901300 longevity-50gb-12h-master-db-node-a991fff7-1 scylla[9237]:  [shard 0: gms] raft_topology - entered `tablet migration` transition state
...
```
and so on

If it happened 1000x times, that would mean the condition for this to occur held for 1000s = 16 minutes

did the test keep some node down for 16 minutes?

Also @tgrabiec is it expected for tablet load balancer to just keep trying like this even though it keeps failing?

**Comment by tgrabiec** [comment]
> Also @tgrabiec is it expected for tablet load balancer to just keep trying like this even though it keeps failing?

We are supposed to retry so that we recover when the node is up again, but not in a tight loop, there should be a 1s sleep between retries.

**Comment by kbr-scylla** [comment]
There is. (But the node was down for 16 minutes, so the log got spammed nevertheless)

**Comment by tgrabiec** [comment]
As an optimization we could freeze and wake up when node UP status changes. But then we'll have to debug missed notification problems.

**Comment by kbr-scylla** [comment]
Alternatives:
- apply log limiter to these particular log messages
- exponential backoff (up to some limit) in the retry loop (but we need to make sure that if something other work appears, not the one we've been retrying since last few minutes, we handle it immediately)
